spdx-utils/src/main/kotlin/SpdxExpression.kt

```diff
@@ -236,6 +253,37 @@ class SpdxCompoundExpression(
             SpdxOperator.AND -> left.offersChoice() || right.offersChoice()
         }
 
+    override fun applyChoice(subtreeMatcher: SpdxExpression, choice: SpdxExpression): SpdxExpression {
+        if (!subtreeMatcher.isValidChoice(choice)) {
+            throw InvalidLicenseChoiceException(
+                "$choice is not a valid choice for the subtree expression. Valid choices are: ${validChoices()}."
+            )
+        }
+
+        if (subtreeMatcher.validChoices().size > 2) {
+            throw InvalidLicenseChoiceException(
+                "Subtree matcher $subtreeMatcher must not contain more than two choices."
+            )
+        }
+
+        return replaceMatcherWithChoice(subtreeMatcher, choice)
+    }
+
+    private fun replaceMatcherWithChoice(subtreeMatcher: SpdxExpression, choice: SpdxExpression): SpdxExpression {
+        val expressionString = toString()
+        val matcherString = subtreeMatcher.toString()
+        val choiceString = choice.toString()
+
+        return if (expressionString.contains(matcherString)) {
+            expressionString.replace(matcherString, choiceString).toSpdx()
+        } else {
+            val dismissedLicense = subtreeMatcher.validChoices().first { it != choice }
+            val remains = this.validChoices().filter { it != dismissedLicense }
```

Comment:
Drop `this.`. Can you maybe think of a better variable name than `remains`?

Reply:
I tried ^^

Reply:
Removed the this. Also found an IntelliJ inspection to warn about redundant `this`.